magento/magento2
# [Backport] Fix the issue with "Shipping address is not set" exception

Original PR: https://github.com/magento/magento2/pull/16753

The exception is being thrown instead of redirecting the customer to New Shipping Address form.

### Description
The issue appeared because of the interrupted shipment assignment process.
As result, the incorrectly assigned record appears in ``quote_address`` table, which causes the additional error on Shopping Cart page, after performing the steps mentioned in *Manual testing scenarios*.
The issue has been fixed by replacing the ``StateException`` throw with void return statement in 
``\Magento\Quote\Model\ShippingMethodManagement::apply`` method. This method is being called by ``\Magento\Quote\Model\Quote\ShippingAssignment\ShippingProcessor::save``.

### Fixed Issues 
1. magento/magento2#16555: "Shipping address is not set" exception in Multishipping Checkout.

### Manual testing scenarios
1. Log In as Customer.
2. Add Product to Cart.
3. Proceed to Checkout.
4. Fill Shipping Address data, click Next (do not place Order)
5. Return to the Storefront.
6. Go to the Shopping Cart.
7. Click Check Out with Multiple Addresses.

### Contribution checklist
